### PR TITLE
Adding environment for MITxPro

### PR DESCRIPTION
**Pull request by blarghmatey**

### CODE REVIEW

File: salt/environment_settings.yml [hunk 1]
```diff
@@ -89,6 +89,70 @@ sandbox_residential_versions: &sandbox_residential_versions
...
+          - ecommerce
+    backends:
+      pki:
+        - consul
+        - elasticsearch
+        - fluentd
+        - mongodb
+        - rabbitmq
+      rds:
+        - name: mitxcas
```
Comment (shaidar) [inline salt/environment_settings.yml]: ?

File: salt/environment_settings.yml [hunk 2]
```diff
@@ -89,6 +89,70 @@ sandbox_residential_versions: &sandbox_residential_versions
...
+          allocated_storage: 50
+          public_access: False
+          engine_version: '10.1'
+          purpose: mitx-cas
+      elasticache:
+        - engine: memcached
+          engine_version: '1.4.34'
+          node_type: cache.t2.small
+          num_cache_nodes: 1
+          cluster_id: continuous-delivery
```
Comment (shaidar) [inline salt/environment_settings.yml]: ?

File: salt/environment_settings.yml [hunk 3]
```diff
@@ -89,6 +89,70 @@ sandbox_residential_versions: &sandbox_residential_versions
...
+          purpose: continuous-delivery
+        - engine: memcached
+          engine_version: '1.4.34'
+          node_type: cache.t2.small
+          num_cache_nodes: 1
+          cluster_id: current-res-live
```
Comment (shaidar) [inline salt/environment_settings.yml]: ?

File: salt/environment_settings.yml [hunk 4]
```diff
@@ -89,6 +89,64 @@ sandbox_residential_versions: &sandbox_residential_versions
...
+      rds:
+        - name: mitxpro-qa
+          multi_az: False
+          engine: mariadb
+          vault_plugin: mysql-database-plugin
+          db_instance_class: db.t2.medium
+          allocated_storage: 100
+          public_access: False
+          engine_version: '10.3'
+          purpose: xpro
```
Comment (shaidar) [inline salt/environment_settings.yml]: Should this be xpro-qa?

File: pillar/edx/ansible_vars/ecommerce.sls [hunk 5]
```diff
@@ -58,15 +58,15 @@ edx:
...
-    ECOMMERCE_CYBERSOURCE_ACCESS_KEY: 'SET-ME-PLEASE'
-    ECOMMERCE_CYBERSOURCE_SECRET_KEY: 'SET-ME-PLEASE'
-    ECOMMERCE_CYBERSOURCE_SOP_ACCESS_KEY: 'SET-ME-PLEASE'
-    ECOMMERCE_CYBERSOURCE_SOP_PROFILE_ID: 'SET-ME-PLEASE'
-    ECOMMERCE_CYBERSOURCE_SOP_SECRET_KEY: 'SET-ME-PLEASE'
+    ECOMMERCE_CYBERSOURCE_PROFILE_ID: __vault__::secret-{{ business_unit }}/{{ environment }}/cybersource>data>profile_id
+    ECOMMERCE_CYBERSOURCE_MERCHANT_ID: __vault__::secret-{{ business_unit }}/{{ environment }}/cybersource>data>merchant_id
+    ECOMMERCE_CYBERSOURCE_ACCESS_KEY: __vault__::secret-{{ business_unit }}/{{ environment }}/cybersource>data>access_key
+    ECOMMERCE_CYBERSOURCE_SECRET_KEY: __vault__::secret-{{ business_unit }}/{{ environment }}/cybersource>data>secret_key
+    ECOMMERCE_CYBERSOURCE_SOP_ACCESS_KEY: __vault__::secret-{{ business_unit }}/{{ environment }}/cybersource>data>access_key
```
Comment (shaidar) [inline pillar/edx/ansible_vars/ecommerce.sls]: Is this supposed to be the same as `ECOMMERCE_CYBERSOURCE_ACCESS_KEY`?

Comment (blarghmatey) [reply pillar/edx/ansible_vars/ecommerce.sls]: For now, yes. At least until we figure out what breaks :)

**Comment by shaidar** [comment]
👍

**Pull request merged**